common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java

```diff
@@ -129,18 +130,27 @@ private String getCredentialType(@NonNull final String tokenType) {
      * @param responseScope The response scope to parse.
      * @return The target containing default scopes.
      */
-    private String getTarget(@NonNull final String responseScope) {
-        if (responseScope.contains(AuthenticationConstants.OAuth2Scopes.OPEN_ID_SCOPE)) {
-            if (responseScope.contains(AuthenticationConstants.OAuth2Scopes.OFFLINE_ACCESS_SCOPE)) {
-                return responseScope;
-            } else {
-                return responseScope + " " + AuthenticationConstants.OAuth2Scopes.OFFLINE_ACCESS_SCOPE;
+    private String getTarget(@NonNull final String requestScope,
+                             @NonNull final String responseScope) {
+        String scopesToCache = "";
+
+        if (TextUtils.isEmpty(responseScope)) {
+            // The response scopes were empty -- per https://tools.ietf.org/html/rfc6749#section-3.3
+            // we are going to fall back to a the request scopes minus any default scopes....
+            final String[] requestScopes = requestScope.split("\\s+");
+            final Set<String> requestScopeSet = new HashSet<>(Arrays.asList(requestScopes));
+            requestScopeSet.removeAll(DEFAULT_SCOPES);
+
+            for (final String scope : requestScopeSet) {
+                scopesToCache += scope + " ";
```

Comment:
I think it would be better to use a `StringBuilder` here as the size of the scope set isn't known and could be big

Reply:
The perf of `+` and `StringBuilder` is fairly comparable due to compiler optimizations. Given that `StringBuilder` hurts readability IMO, I'm going to leave as is.

(Citation: [RedFin Engineering Blog](https://redfin.engineering/java-string-concatenation-which-way-is-best-8f590a7d22a8))
![Capture](https://user-images.githubusercontent.com/990063/73968807-25ede800-48cf-11ea-9585-caed23c288d3.PNG)